packages/terra-tabs/src/Tabs.jsx

```diff
@@ -0,0 +1,226 @@
+import TabPane from './TabPane';
+import CollapsibleTabs from './_CollapsibleTabs';
+import CollapsedTabs from './_CollapsedTabs';
+import TabUtils from './TabUtils';
+import styles from './Tabs.scss';
+
+const cx = classNames.bind(styles);
+
+/**
+NOTE: This is being commented out until descussions have been resolved around if modular tabs should be removed.
+const variants = {
+  MODULAR_CENTERED: 'modular-centered',
+  MODULAR_LEFT_ALIGNED: 'modular-left-aligned',
+  STRUCTURAL: 'structural',
+};
+*/
+
+const propTypes = {
+
+  /**
+   * Tabs style. One of: Tabs.Opts.Variants.MODULAR_CENTERED, Tabs.Opts.Variants.MODULAR_LEFT_ALIGNED, or Tabs.Opts.Variants.STRUCTURAL.
+   * NOTE: This is being commented out until descussions have been resolved around if we want modular tabs.
+  variant: PropTypes.oneOf([variants.MODULAR_CENTERED, variants.MODULAR_LEFT_ALIGNED, variants.STRUCTURAL]),
+  */
+
+  /**
+   * Indicates if tabs should fill the width available in the tab bar.
+   */
+  tabFill: PropTypes.bool,
+
+  /**
+   * Indicates if the pane content should fill to the height of the parent container.
+   */
+  fill: PropTypes.bool,
+
+  /**
+   * Callback function when selection has changed.
+   * Parameters: 1. Event 2. Selected pane's key
+   */
+  onChange: PropTypes.func,
+
+  /**
+   * Tabs.Pane components to be displayed
+   */
+  children: PropTypes.node.isRequired,
+
+  /**
+   * Key of the pane that should be active. Use this prop along with onChange to create controlled tabs.
+   */
+  activeKey: PropTypes.string,
+
+  /**
+   * Key of the pane that should be open initially.
+   */
+  defaultActiveKey: PropTypes.string,
+};
+
+const defaultProps = {
+  tabFill: false,
+  fill: false,
+};
+
+class Tabs extends React.Component {
+  constructor(props) {
+    super(props);
+    this.getInitialState = this.getInitialState.bind(this);
+    this.getActiveTabIndex = this.getActiveTabIndex.bind(this);
+    this.handleOnChange = this.handleOnChange.bind(this);
+    this.handleTruncationChange = this.handleTruncationChange.bind(this);
+    this.wrapPaneOnClick = this.wrapPaneOnClick.bind(this);
+    this.state = {
+      activeKey: this.getInitialState(),
+      isLabelTruncated: false,
+    };
+  }
+
+  getInitialState() {
+    if (this.props.activeKey) {
+      return null;
+    }
+    return TabUtils.initialSelectedTabKey(this.props.children, this.props.defaultActiveKey);
+  }
+
+  getActiveTabIndex() {
+    let activeIndex = -1;
+    React.Children.forEach(this.props.children, (child, index) => {
+      if (child.key === this.state.activeKey || child.key === this.props.activeKey) {
+        activeIndex = index;
+      }
+    });
+
+    return activeIndex;
+  }
+
+
+  handleOnChange(event, selectedPane) {
+    if (!selectedPane.props.isDisabled) {
+      if (this.props.onChange) {
+        this.props.onChange(event, selectedPane.key);
+      } else if (TabUtils.shouldHandleSelection(this.state.activeKey, selectedPane.key)) {
```

Comment:
Should this be an if-else here? I'm assuming this is laid out for controlled vs uncontrolled component, but I am wondering if in both scenarios we'd honor onChange and then check if this.props.activeKey is provided before updating state.